Create 2 GitHub action runners to run tests

HonkingGoose opened the pull request:
## Changes:

- Creates 2 new GitHub Action Node.js test runners

## Context:

Travis CI is moving to a `.com` domain.
You no longer get free tests for public repositories.
You therefore might want to migrate away from Travis CI and to GitHub Actions.
GitHub actions are free for public repositories.

`npm test` runs on: Node 6, 8, 10 and 14.
`npm run testv12` runs on Node 12.

Tests are all passing now.

After you have reviewed and merged this PR you can remove the Travis CI config file, if you want.

HonkingGoose commented:
Hmm the GitHub actions are not running on this repository.

You can see the current tests in action at my fork: https://github.com/HonkingGoose/http-parser-js/pull/1

Jimbly commented:
Thanks for the contribution, and the heads-up!

Did a bunch of debugging, and it looks like the tests broke on Node v12.19.0, with the addition of this line to Node.js:
```javascript
const kOnTimeout = HTTPParser.kOnTimeout | 0;
```
Which is treating a new, unknown constant as "0", which is the same index that's used for `kOnHeaders`, so it's blowing away the header handling callback ^_^.  I pushed a fix on our side, and we'll not use 0 for an ID ever again, even if that's what the legacy http parser uses.  I guess merge upstream into your branch and check if your GitHub actions pass now?

HonkingGoose commented:
> Thanks for the contribution, and the heads-up!

Cheers! 😄 

>  I guess merge upstream into your branch and check if your GitHub actions pass now?

I'm current with upstream, the Node 12 test suite now works.
The Node 14 and 15 tests still fail.

Node 14 log: https://github.com/HonkingGoose/http-parser-js/pull/1/checks?check_run_id=1634328734
Node 15 log: https://github.com/HonkingGoose/http-parser-js/pull/1/checks?check_run_id=1634328744

HonkingGoose commented:
Oh by the way, I can also get you a **Dependabot** config to update the GitHub action runners.
That way you're always using the latest versions of the runners (be aware that Dependabot will update to unstable versions of the runners though...).

Dependabot is the GitHub-native dependency manager bot.

You can read the official GitHub docs for updating GitHub Actions with Dependabot here: https://docs.github.com/en/free-pro-team@latest/github/administering-a-repository/keeping-your-actions-up-to-date-with-dependabot#about-dependabot-version-updates-for-actions

Let me know if you'd like that Dependabot config or not. I can just add it to this PR. 😉

Jimbly commented:
I fixed the tests on Node v14/15 (test problems, not parser problems).  I don't think we want to test against Node v15 on PRs though - if only that version fails, it's almost certainly a problem due to Node's internals changing (expected, as we're not using "supported" APIs here), not due to anything that a PR would be changing, and often requires quite a bit of investigation to resolve (just like today, which has nothing to do with moving to GitHub Actions ^_^).

Also, we should just run `npm test` on all versions except for v12 (Node v14 is tested with `npm test` and later Node versions will probably error with the `--http-parser` option that is now deprecated).  Maybe name the actions `node` and `node-v12` instead of `old` and `new`?

HonkingGoose commented:
> I fixed the tests on Node v14/15 (test problems, not parser problems).

All tests pass now. 👍 

>  I don't think we want to test against Node v15 on PRs though

Removed Node 15 from the test matrix.

> run `npm test` on all versions except for v12

Fixed.

> Maybe name the actions `node` and `node-v12` instead of old and new?

Fixed.

HonkingGoose commented:
I think this is ready for a final review.

I'll stop changing things now. 😄

Jimbly commented:
Looks good, thanks!
